Package.swift

```diff
@@ -2,23 +2,64 @@
 
 import PackageDescription
 
+let MeiliSearch = "MeiliSearch"
+let MeiliSearchCore = "MeiliSearchCore"
+let MeiliSearchNIO = "MeiliSearchNIO"
+
+var products: [PackageDescription.Product] = [
+  .library(name: MeiliSearch, targets: [MeiliSearch]),
+  .library(name: MeiliSearchCore, targets: [MeiliSearchCore])
+]
+
+var dependencies = [PackageDescription.Package.Dependency]()
+
+var targets: [PackageDescription.Target] = [
+  .target(
+    name: MeiliSearch,
+    dependencies: [
+      .target(name: MeiliSearchCore)
+    ]
+  ),
+  .target(
+    name: MeiliSearchCore, 
+    dependencies: []
+  ),
+  .testTarget(
+    name: "MeiliSearchUnitTests",
+    dependencies: ["MeiliSearch"]
+  ),
+  .testTarget(
+    name: "MeiliSearchIntegrationTests",
+    dependencies: ["MeiliSearch"]
+  )
+]
+
+#if os(Linux) || USE_NIO
+products.append(.library(name: MeiliSearchNIO, targets: [MeiliSearchNIO]))
+dependencies.append(.package(url: "https://github.com/swift-server/async-http-client.git", from: "1.19.0"))
+targets.append(
+  .target(
+    name: MeiliSearchNIO,
```

Comment:
major: this PR will currently fail to build as-is due to this target path not existing. shall we delete this entire Linux section until you contribute the NIO integration in your next PR?